mbslave/replication.py

```
@@ -238,44 +250,119 @@ def check_table_exists(db, schema, table):
+    work_dir = args.work_dir or os.getcwd()
+    if not os.path.exists(work_dir):
+        os.makedirs(work_dir)
+
     for source in args.sources:
-        with ExitStack() as exit_stack:
-            if source.startswith('http://') or source.startswith('https://'):
-                fileobj = exit_stack.enter_context(urlopen(source))
-            else:
-                fileobj = exit_stack.enter_context(open(source, 'rb'))
-            load_tar(source, fileobj, db, config, config.schemas.ignored_schemas, config.tables.ignored_tables)
+        if source.startswith('http://') or source.startswith('https://'):
```

Ruff (0.14.10)

Call `startswith` once with a `tuple`

Merge into a single `startswith` call

(PIE810)